src/components/30-organisms/carousel/index.js

```diff
@@ -0,0 +1,228 @@
+import { LitElement, html, css, unsafeCSS } from 'lit-element';
+
+/* eslint-disable import/no-extraneous-dependencies */
+import defineOnce from '../../../utils/define-once';
+import styles from './index.scss';
+import Swipe from './swipe';
+import debounce from '../../../utils/debounce';
+
+const ELEMENT_NODE = 1;
+const ANIMATION_LEFT_CLASS = 'animation-left';
+const ANIMATION_RIGHT_CLASS = 'animation-right';
+
+class AXACarousel extends LitElement {
+  static get tagName() {
+    return 'axa-carousel';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    return {
+      autorotatedisabled: { type: Boolean },
+      autorotatetime: { type: Number },
+      keysenabled: { type: Boolean },
+      _animationWrapperClass: { type: String },
+      _carouselMinHeight: { type: Number },
+    };
+  }
+
+  // public methods
+
+  handleNextButtonClick() {
+    this._nextSlide();
+    this._stopAutoRotate();
+  }
+
+  handlePreviousButtonClick() {
+    this._previousSlide();
+    this._stopAutoRotate();
+  }
+
+  // private methods
+
+  _getSlides(usedShadowRoot = this.shadowRoot) {
+    // usedShadowRoot as a property to encapsulate for unit tests
+    const slots = usedShadowRoot.querySelector('.o-carousel__slot');
+
+    return slots
+      .assignedNodes({ flatten: true })
+      .filter(node => node.nodeType === ELEMENT_NODE);
+  }
+
+  _setSlideVisibleWithAnimation(slideIndex, animationClass) {
+    this.visibleSlideIndex = slideIndex;
+    this._animationWrapperClass = '';
+    this.slides.forEach(node => {
+      node.style.display = 'none';
+    });
+
+    setTimeout(() => {
+      // Browser needs time to render css animation (>0ms). Other workarounds doesnt have effect.
+      this._animationWrapperClass = animationClass;
+      this.slides[slideIndex].style.display = 'block';
+    }, 50);
```

Comment:
This is always tricky. How do you know when css animation is done? either having same timeout as in the scss file or by listening to callback. Here the danger is that if u change the animation in css u absolutely have to not forget this number. Please either CROSS reference the files OR listen to transitionend: https://www.w3schools.com/jsref/event_animationend.asp

Reply:
And VERY IMPORTANT clear timeout before

Reply:
This timeout time is not the animation time. This time is just for the browser to rerender, because i am removing a class and set this class in the same tick.

We have 3 options:
- timeout > 0ms
- event-listener on animation (too heavy)
- a render hack (not working for us)

Reply:
resolved